src/diffusers/models/upsampling.py

```diff
@@ -165,6 +165,14 @@ def forward(self, hidden_states: torch.Tensor, output_size: Optional[int] = None
         # if `output_size` is passed we force the interpolation output
         # size and do not make use of `scale_factor=2`
         if self.interpolate:
+            # upsample_nearest_nhwc also fails when the number of output elements is large
```

Comment:
Do we need to check for `channels_first` or `channels_last` memory layout?

Reply:
My assumption is it will behave the same way for any non-contiguous layouts, and that calling `.contiguous()` on a contiguous tensor will be a no-op. I'm also mimicking the same structure as in [lines 161-163 above](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/upsampling.py#L161-L163).